go_to_office/lib/screens/login.dart

```diff
@@ -19,52 +20,111 @@ class _LoginPageState extends State<LoginPage> {
   _LoginPageState(this.repository);
   final Repository repository;
 
+  void onUserLoggedIn() {
+    Navigator.push(
+      context,
+      MaterialPageRoute(
+          builder: (context) => Meetings(
+            title: "Meetings",
+          )),
+    );
+  }
+
+  void onAdminLoggedIn() {
+    Navigator.push(
+      context,
+      MaterialPageRoute(builder: (context) => OfficesListPage(),
+    ));
+  }
+
   final _userTextController = TextEditingController();
   final _passTextController = TextEditingController();
+  final GlobalKey<ScaffoldState> _scaffoldKey = new GlobalKey<ScaffoldState>();
 
   @override
   void dispose() {
-    // Clean up the controller when the widget is disposed.
+    // Cleans up the controller when the widget is disposed.
     _userTextController.dispose();
     _passTextController.dispose();
     super.dispose();
   }
 
   @override
   Widget build(BuildContext context) {
-    void showSnack(String text) {
-      final snackBar = SnackBar(content: Text(text));
-      // TBC -> implement it by Oz
+
+    void _showSnack(String text) {
+      _scaffoldKey.currentState.showSnackBar( SnackBar(content: Text(text)) );
+    }
+
+    void _showError(String text) {
+      _scaffoldKey.currentState.showSnackBar( SnackBar(content: Text(text), backgroundColor: Colors.red) );
+    }
+
+    void _onError(Object error) {
+      String text = "";
+      if (error is FirebaseAuthException) {
+        text = error.message;
+      } else {
+        text = error.toString();
+      }
+      _showError(text);
     }
 
-    void onError(String text) {
-      final snackBar =
-          SnackBar(content: Text(text), backgroundColor: Colors.red);
-      // TBC -> implement it by Oz
+    bool _userCredentialsValidation(String _email, String _password) {
+
+      bool _validPassword = _password.isNotEmpty && _password.length > 5;
+      bool _validEmail = _email.isNotEmpty && RegExp(r"^[a-zA-Z0-9.a-zA-Z0-9.!#$%&'*+-/=?^_`{|}~]+@[a-zA-Z0-9]+\.[a-zA-Z]+").hasMatch(_email);
+
+      if (!_validEmail) {
+        _onError(Strings.invalid_email);
+        return false;
+      }
+
+      if (!_validPassword) {
+        _onError(Strings.invalid_password);
+        return false;
+      }
+
+      return true;
+    }
+
+    void _dispatchLogin(String pressedOption, String uid) {
+
+      // /*uncomment to present a snack*/ _showSnack("User had been successfully logged in" + uid);
+
+      if (pressedOption == Strings.loginUser)
+        onUserLoggedIn();
+      else
+        onAdminLoggedIn();
+
     }
 
-    void _onLoginRequest(BuildContext context, String pressedOption) {
-      if (pressedOption == Strings.loginUser) {
+    /// If the user is not registered, this method registers the user and then sign him in.
+    void _onLoginRequest(String pressedOption) {
+
+      String _email = _userTextController.text.toString();
+      String _password = _passTextController.text.toString();
+
+      if (_userCredentialsValidation(_email, _password)) {
```

Comment:
Do you have a place where you check if the _email and _password are not empty before calling this function?
if not - then this function should check for empty values